core/src/main/java/org/ldaptive/pool/AbstractConnectionPool.java

```diff
@@ -585,33 +585,40 @@ protected void grow(final int size)
    */
   protected void grow(final int size, final boolean throwOnFailure)
   {
-    logger.trace("waiting for pool lock to initialize pool {}", poolLock.getQueueLength());
-
-    int count = 0;
-    poolLock.lock();
-    try {
-      IllegalStateException lastThrown = null;
-      int currentPoolSize = active.size() + available.size();
-      logger.debug("Checking connection pool size >= {} for {}", size, this);
-      while (currentPoolSize < size && count < size * 2) {
+    if (checkOutLock.tryLock()) {
+      try {
+        int count = 0;
+        logger.trace("waiting for pool lock to initialize pool {}", poolLock.getQueueLength());
+        poolLock.lock();
         try {
-          final PooledConnectionProxy pc = createAvailableConnection(throwOnFailure);
-          if (pc != null && connectOnCreate) {
-            if (!passivateAndValidateConnection(pc)) {
-              removeAvailableConnection(pc);
+          IllegalStateException lastThrown = null;
+          int currentPoolSize = active.size() + available.size();
```

Comment:
Seems useful but I've avoided it because documenting the usage around locking semantics would be tricky. That's because in most cases you want the answer to that question while holding the `poolLock` so you can do something to the pool without the answer changing underneath you. Certainly a solvable problem it just always seemed to me that it would add additional complexity to an already complex implementation.